src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/service/zookeeper/ZooKeeperService.java

```diff
@@ -0,0 +1,132 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.service.zookeeper;
+
+import org.openkilda.bluegreen.LifeCycleObserver;
+import org.openkilda.bluegreen.LifecycleEvent;
+import org.openkilda.bluegreen.Signal;
+import org.openkilda.bluegreen.ZkStateTracker;
+import org.openkilda.bluegreen.ZkWatchDog;
+import org.openkilda.bluegreen.ZkWriter;
+import org.openkilda.floodlight.KafkaChannel;
+import org.openkilda.floodlight.service.IService;
+import org.openkilda.floodlight.service.kafka.KafkaUtilityService;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import net.floodlightcontroller.core.module.FloodlightModuleException;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import net.jodah.failsafe.SyncFailsafe;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class ZooKeeperService implements IService, LifeCycleObserver {
+
+    public static final String ZK_COMPONENT_NAME = "floodlight";
+    public static final int ZK_CONNECTION_ATTEMPTS = 10;
+
+
+    private final Set<ZooKeeperEventObserver> observers = new HashSet<>();
+
+    @Getter
+    private ZkStateTracker zooKeeperStateTracker;
+    private ZkWriter zkWriter;
+    private ZkWatchDog watchDog;
+
+    private LifecycleEvent event;
+
+    private long messageId = 0;
+
+    @Override
+    public void setup(FloodlightModuleContext moduleContext) throws FloodlightModuleException {
+        KafkaChannel kafkaChannel = moduleContext.getServiceImpl(KafkaUtilityService.class).getKafkaChannel();
+        String region = kafkaChannel.getRegion();
+        String connectionString = kafkaChannel.getConfig().getZooKeeperConnectString();
+        zkWriter = ZkWriter.builder().id(region).serviceName(ZK_COMPONENT_NAME)
+                .connectionString(connectionString).build();
+        zooKeeperStateTracker = new ZkStateTracker(zkWriter);
+
+        watchDog = ZkWatchDog.builder().id(region).serviceName(ZK_COMPONENT_NAME)
+                .connectionString(connectionString).build();
+        watchDog.subscribe(this);
+    }
+
+    /**
+     * Connects to zookeeper.
+     */
+    public void initZookeeper() {
+        zkWriter.init();
+        watchDog.init();
+
+        RetryPolicy retryPolicy = new RetryPolicy()
+                .retryOn(IllegalStateException.class)
+                .withDelay(100, TimeUnit.MILLISECONDS)
+                .withMaxRetries(ZK_CONNECTION_ATTEMPTS);
```

Comment:
we want to add force read after connection. Are we sure that 10 attempts is enough? I don't know how may retries we need. But in Interceptors we have infinite retries